ob1k-cache/src/main/java/com/outbrain/ob1k/cache/memcache/MemcacheClient.java

```diff
@@ -31,15 +38,38 @@
  */
 public class MemcacheClient<K, V> implements TypedCache<K, V> {
   private static final long MAX_EXPIRATION_SEC = 60 * 60 * 24 * 30;
+  private static final String COMPONENT = MemcacheClient.class.getSimpleName();
 
   private final MemcachedClientIF spyClient;
   private final CacheKeyTranslator<K> keyTranslator;
   private final int expirationSpyUnits;
 
-  public MemcacheClient(final MemcachedClientIF spyClient, final CacheKeyTranslator<K> keyTranslator, final long expiration, final TimeUnit timeUnit) {
+  private MetricFactory metricFactory = null;
```

Comment:
ok - corrected and updated commit:
https://github.com/outbrain/ob1k/pull/25/files

On Sun, Nov 29, 2015 at 9:14 AM, Haim Yadid <notifications@github.com>
wrote:

> In
> ob1k-cache/src/main/java/com/outbrain/ob1k/cache/memcache/MemcacheClient.java
> <https://github.com/outbrain/ob1k/pull/25#discussion_r46088495>:
>
> >
> >    private final MemcachedClientIF spyClient;
> >    private final CacheKeyTranslator<K> keyTranslator;
> >    private final int expirationSpyUnits;
> >
> > -  public MemcacheClient(final MemcachedClientIF spyClient, final CacheKeyTranslator<K> keyTranslator, final long expiration, final TimeUnit timeUnit) {
> > +  private MetricFactory metricFactory = null;
>
> I agree with Michael on this. Keeping the metric factory and relying on it
> to check other members is error prone. The minor "complexity" created is
> cancelled by benefits of locality
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/outbrain/ob1k/pull/25/files#r46088495>.
>